mcc/web/file_utils.py

```diff
@@ -156,7 +156,11 @@ def get_dataset_from_file(uploaded_file):
     # to read the file contents but not reference a location on disk.
     # Copy the uploaded file to a named temporary file, so we can provide a disk
     # location when initializing the netCDF.Dataset object below.
-    datafile = tempfile.NamedTemporaryFile()
+
+    # Since the CF suite checks the Dataset's filename for compliance,
+    # we need to make sure the original is incorporated into the name of the
+    # temporary file used to initialize the Dataset object below.
+    datafile = tempfile.NamedTemporaryFile(suffix=f"_{datafile_name}")
```

Comment:
If the goal is to preserve the original filename, should we consider using a TemporaryDirectory instead and manually create a file object from there? Then once the TemporaryDirectory is closed/goes out of scope, the file gets cleaned up along with it (assuming it's been closed as well)

Reply:
The goal was actually to preserve the file extension, which is what the CF convention actually cares about. This seemed sufficient to achieve that.